test/unit/org/apache/cassandra/utils/jmx/JMXSslDisabledEncryptionOptionsTest.java

```diff
@@ -0,0 +1,110 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils.jmx;
+
+import java.net.InetAddress;
+import java.util.Map;
+import javax.management.remote.rmi.RMIConnectorServer;
+import javax.net.ssl.SSLException;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.distributed.shared.WithProperties;
+import org.apache.cassandra.utils.JMXServerUtils;
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.CASSANDRA_CONFIG;
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL;
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_CIPHER_SUITES;
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_PROTOCOLS;
+import static org.apache.cassandra.config.CassandraRelevantProperties.COM_SUN_MANAGEMENT_JMXREMOTE_SSL_NEED_CLIENT_AUTH;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_RMI_SSL_CLIENT_ENABLED_CIPHER_SUITES;
+import static org.apache.cassandra.config.CassandraRelevantProperties.JAVAX_RMI_SSL_CLIENT_ENABLED_PROTOCOLS;
+
+/**
+ * Tests for disabling jmx_encryption_options in the cassandra.yaml.
+ */
+public class JMXSslDisabledEncryptionOptionsTest
+{
+    static WithProperties properties;
+
+    @BeforeClass
+    public static void setupDatabaseDescriptor()
+    {
+        properties = new WithProperties().set(CASSANDRA_CONFIG, "cassandra-jmx-disabled-sslconfig.yaml");
+        DatabaseDescriptor.daemonInitialization();
+    }
+
+    @AfterClass
+    public static void tearDownDatabaseDescriptor()
+    {
+        properties.close();
+    }
+
+    @After
+    public void resetJmxSslSystemProperties()
+    {
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL.reset();
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_NEED_CLIENT_AUTH.reset();
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_PROTOCOLS.reset();
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_CIPHER_SUITES.reset();
+        JAVAX_RMI_SSL_CLIENT_ENABLED_PROTOCOLS.reset();
+        JAVAX_RMI_SSL_CLIENT_ENABLED_CIPHER_SUITES.reset();
+    }
+
+    /**
+     * Tests absence of all JMX SSL configurations,
+     * 1. local only JMX server
+     * 2. System properties set for remote JMX SSL
+     * 3. jmx_encryption_options in the cassandra.yaml
+     */
+    @Test
+    public void testAbsenceOfAllJmxSslConfigs() throws SSLException
```

Comment:
The comments and the setup don't really match in this case - the _absense_ of all JMX SSL configuration should be no JMX config in cassandra.yaml, right? But this class loads a config that has those settings, but disabled.

Reply:
The challenge is because `jmx_encryption_options` get defaults when we don't specify it in cassandra.yaml. So I meant to say that - this test validates use-case when none of the below are configured-

1. JMX is turned off with `com.sun.management.jmxremote.ssl` system property
2. system configuration for the JMX
3. `jmx_encryption_options` in cassandra.yaml

I have tried to rename the method and update the documentation. Please review if the next change makes more sense.